codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieHeaderNames.java

```diff
@@ -28,6 +28,39 @@
 
     public static final String HTTPONLY = "HTTPOnly";
 
+    public static final String SAMESITE = "SameSite";
+
+    /**
+     * Possible values for the SameSite attribute.
+     * See <a href=https://tools.ietf.org/html/draft-west-cookie-incrementalism-00#section-3.1>changes to RFC6265bis</a>
+     */
+    public enum SameSite {
+        Lax,
+        Strict,
+        None;
+
+        /**
+         * Return the enum value corresponding to the passed in same-site-flag, using a case insensitive comparison.
+         *
+         * @param name value for the SameSite Attribute
+         * @return enum value for the provided name or null
+         */
+        static SameSite of(String name) {
+            if (name != null) {
+                try {
+                    return SameSite.valueOf(name);
+                } catch (IllegalArgumentException e) {
```

Comment:
I'm really not fond of using exceptions for expected flow control.
IMHO, as value matching must be case insensitive, we should drop the SameSite.valueOf and have one single code branch.

Reply:
I agree with @slandelle here @dvlato

Reply:
In that case, should I remove the logging altogether in ClientCookieDecoder?

Reply:
@dvlato imho we can remove the logging and the `valueOf` alltogether and just do the for loop directly. I think this is what @slandelle suggested